File: SelectionSort.cpp
// ConsoleApplication1.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include <fstream>
#include <iostream>
#include <sstream>
using namespace std;

int _tmain(int argc, _TCHAR* argv[])
{
	// Step 0
	int dataCount = 0;
	string filename, outGoing;
	int* dataSorted;
	int startIndex, endIndex, walkerIndex, minIndex, temp;
	ifstream inFile;
	ofstream outFile;

	// Step 1
	cout << "Please provide a filename: ";
	getline(cin,filename);
	inFile.open(filename);

	// Step 2
	int takeOut = 0;
	string line;
	while (getline(inFile,line)) {
		istringstream iss(line);
		while (iss >> takeOut) dataCount++;
	}

	// Step 3
	inFile.close();

	// Step 4
	dataSorted = new int[dataCount];

	// Step 5
	inFile.open(filename);

	// Step 6
	for (int i = 0; i < dataCount; i++) {
		while (getline(inFile,line)) {
			istringstream iss(line);
			while (iss >> takeOut) { 
				dataSorted[i] = takeOut;
				i++;
			}
		}
	}

	// Step 7
	startIndex = 0;
	endIndex = dataCount - 1;

	// Step 8
	while (!(startIndex >= endIndex)) {
		minIndex = startIndex;
		walkerIndex = startIndex + 1;

		// Step 9
		while (!(walkerIndex > endIndex)) {
			if (dataSorted[walkerIndex] < dataSorted[minIndex])
				minIndex = walkerIndex;

			// Step 10
			walkerIndex++;

			// Step 11
			// return to whileLoop
		}

		// Step 12
		temp = dataSorted[startIndex];
		dataSorted[startIndex] = dataSorted[minIndex];
		dataSorted[minIndex] = temp;

		// Step 13
		startIndex++;

		// Step 14
		// return to whileLoop
	}

	// Step 15
	cout << "Please provide an output file name: ";
	getline(cin,outGoing);
	outFile.open(outGoing);

	// Step 16
	for (int j = 0; j < dataCount; j++)
		outFile << dataSorted[j] << " ";

	// Closing
	delete[] dataSorted;
	outFile.close();
	inFile.close();
	return 0;
} // SelectionSort

File: SelectionSort.java
import java.io.*;
import java.util.Scanner;

public class SelectionSort {

    public static void main(String[] args) {

        // Step 0
        int dataCount = 0;
        Scanner keyboard = new Scanner(System.in);
        Scanner fromFile;
        String filename, outGoing;
        File input;
        int[] dataSorted;
        int startIndex, endIndex, walkerIndex, minIndex, temp;
        PrintWriter outStream;

        // Step 1
        System.out.print("Please provide a filename: ");
        filename = keyboard.next();

        try {
            // Step 2
            input = new File(filename);
            fromFile = new Scanner(input);
            int takeOut;
            while (fromFile.hasNextInt()) {
                dataCount++;
                takeOut = fromFile.nextInt();
            }

            // Step 3
            fromFile.close();

            // Step 4
            dataSorted = new int[dataCount];

            // Step 5
            // Variable "File input" does not need to change.

            // Step 6
            fromFile = new Scanner(input);
            for (int i = 0; i < dataSorted.length; i++) {
                int store = fromFile.nextInt();
                dataSorted[i] = store;
            }

            // Step 7
            startIndex = 0;
            endIndex = dataCount - 1;

            // Step 8
            while (!(startIndex >= endIndex)) {
                minIndex = startIndex;
                walkerIndex = startIndex + 1;

                // Step 9
                while (!(walkerIndex > endIndex)) {
                    if (dataSorted[walkerIndex] < dataSorted[minIndex])
                        minIndex = walkerIndex;

                    // Step 10
                    walkerIndex++;

                    // Step 11
                    // return to whileLoop
                }

                // Step 12
                temp = dataSorted[startIndex];
                dataSorted[startIndex] = dataSorted[minIndex];
                dataSorted[minIndex] = temp;

                // Step 13
                startIndex++;

                // Step 14
                // return to whileLoop
            }

            // Step 15
            System.out.print("Please provide an output file name: ");
            outGoing = keyboard.next();
            outStream = new PrintWriter(new FileOutputStream(outGoing));

            // Step 16
            for (int j = 0; j < dataSorted.length; j++) {
                outStream.print(dataSorted[j] + " ");
            }
            outStream.close();
            keyboard.close();

        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }

    } // main

} // SelectionSort